sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/ItemComparator.java

```diff
@@ -41,6 +44,17 @@ public int compare(Object obj1, Object obj2) {
             return Double.compare(((Number) obj1).doubleValue(), ((Number) obj2).doubleValue());
         case String:
             return ((String) obj1).compareTo((String) obj2);
+        case ArrayNode:
+        case ObjectNode:
+            try{
+                UInt128 hash1 = DistinctHash.getHash(obj1);
+                UInt128 hash2 = DistinctHash.getHash(obj2);
+                return hash1.compareTo(hash2);
+            }
+            catch (IOException e) {
+                throw new IllegalStateException(String.format("Getting hash exception for type %s ", type1.toString()), e);
```

Comment:
I don't think this exception message makes sense, since there is no `type1` in this try catch block.

Reply:
hmmm, we could remove the type from the exception message, but thought it might give some hint which type is having hash issue?

Reply:
why only type1 and not type2 ?
Also, what does type1.toString() represents ?

Reply:
since we can only reach to this step when type1 equals type2, so I think either should work. 
it will tell us whether it is objectNode or arrayNode I think